Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1/defaults_test.go

```diff
@@ -723,3 +724,77 @@ func TestDefaultSecurityContextConstraints(t *testing.T) {
 		}
 	}
 }
+
+func TestDefaultSCCVolumes(t *testing.T) {
+	tests := map[string]struct {
+		scc             *versioned.SecurityContextConstraints
+		expectedVolumes []versioned.FSType
+		expectedHostDir bool
+	}{
+		// this expects the volumes to default to all for an empty volume slice
+		// but since the host dir setting is false it should be all - host dir
+		"old client - default allow* fields, no volumes slice": {
+			scc:             &versioned.SecurityContextConstraints{},
+			expectedVolumes: versioned.StringSetToFSType(sccutil.GetAllFSTypesExcept(string(versioned.FSTypeHostPath))),
+			expectedHostDir: false,
+		},
+		// this expects the volumes to default to all for an empty volume slice
+		"old client - set allowHostDir true fields, no volumes slice": {
+			scc: &versioned.SecurityContextConstraints{
+				AllowHostDirVolumePlugin: true,
+			},
+			expectedVolumes: []versioned.FSType{versioned.FSTypeAll},
+			expectedHostDir: true,
+		},
```

Comment:
add the following tests:
* empty volumes slice (`[]`), allowHostDir:false, should result in empty slice
* empty volumes slice (`[]`), allowHostDir:true, should result in hostdir slice